src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,155 @@
+        asserts.assert_in(node_label_attr, basic_information, "NodeLabel not in BasicInformation")
+
+        # TH sends the Read Request Message to the DUT to read a global attribute from all clusters at that Endpoint
+        # AttributePath = [[Endpoint = Specific Endpoint, Attribute = Specific Global Attribute]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the clusters to the DUT.
+        self.print_step(4, "Send Request Message to read one global attribute from all clusters at that endpoint")
+        # The output from this command gets many more values from unit_testing_attr compared to the some other tests
+        # [(Clusters.BasicInformation.Attributes.ClusterRevision)]
+        read_request_4 = await self.default_controller.ReadAttribute(self.dut_node_id, cluster_rev_attr_path)
+        attributes = read_request_4[0]
+        basic_information = attributes[basic_info_attr]
+        asserts.assert_true(basic_information[cluster_revision_attr], 3)
+
+        # TH sends the Read Request Message to the DUT to read all attributes from all clusters on all Endpoints
+        ### AttributePath = [[]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the clusters to the DUT.
+        self.print_step(5, "Send Request Message to read all attributes from all clusters on all endpoints")
+        # This returns a dataclass key of 1, unlike 0 for the earlier ones
+        read_request_5 = await self.default_controller.ReadAttribute(self.dut_node_id, [()])
+        attributes = read_request_5[1]
+        unit_testing = attributes[unit_testing_attr]
+
+        # asserts.assert_in(node_label_attr, unit_testing, "NodeLabel not in UnitTesting")
+        asserts.assert_in(data_version_attr, unit_testing, "DataVersion not in UnitTesting")
+        # TH sends the Read Request Message to the DUT to read a global attribute from all clusters at all Endpoints
+        # AttributePath = [[Attribute = Specific Global Attribute]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the clusters to the DUT.
+        self.print_step(6, "Send Request Message to read one global attribute from all clusters on all endpoints")
+        # Clusters.BasicInformation.Attributes.ClusterRevision
+        read_request_6 = await self.default_controller.ReadAttribute(self.dut_node_id, [cluster_rev_attr])
+
+        attributes = read_request_6[0]
+        basic_information = attributes[basic_info_attr]
+        # asserts.assert_in(node_label_attr, basic_information, "NodeLabel not in BasicInformation")
+        asserts.assert_in(data_version_attr, basic_information, "DataVersion not in BasicInformation")
+        asserts.assert_true(basic_information[cluster_revision_attr], 3)
+
+        # TH sends the Read Request Message to the DUT to read all attributes from a cluster at all Endpoints
```

Comment:
use descriptor, which is present on all endpoints

Reply:
Used descriptor for all steps